Tests/MapboxCoreNavigationIntegrationTests/RouteRefreshIntegrationTests.swift

```diff
@@ -0,0 +1,285 @@
+        let stepLastLocation = stepCoordiantes.last!
+        let stepDirection = stepFirstLocation.direction(to: stepLastLocation)
+
+        let offRouteCoordiantes =  [20, 30, 40].map { stepLastLocation.coordinate(at: $0, facing: stepDirection) }
+        let coordinates = stepCoordiantes + offRouteCoordiantes
+        return coordinates
+            .map { CLLocation(latitude: $0.latitude, longitude: $0.longitude) }
+            .shiftedToPresent()
+            .qualified()
+    }
+}
+
+fileprivate extension NavigationRouteOptions {
+    static func mockedOptions(
+        _ profile: ProfileIdentifier
+    ) -> NavigationRouteOptions {
+        NavigationRouteOptions(
+            coordinates: [
+                .origin,
+                .destiantion
+            ],
+            profileIdentifier: profile
+        )
+    }
+
+    static func mockedCustomOptions(
+        _ profile: ProfileIdentifier
+    ) -> NavigationRouteOptions {
+        CustomRouteOptions(
+            waypoints: [
+                .init(coordinate: .origin),
+                .init(coordinate: .destiantion),
+            ],
+            profileIdentifier: profile,
+            customParameters: [.customItem]
+        )
+    }
+
+}
+
+fileprivate extension RouteResponse {
+    static func mockedResponse(profile: ProfileIdentifier) -> RouteResponse {
+        Fixture.routeResponse(
+            from: "profile-route-original",
+            options: NavigationRouteOptions.mockedOptions(profile)
+        )
+    }
+
+    static func mockedIndexRouteResponse(
+        profile: ProfileIdentifier
+    ) -> IndexedRouteResponse {
+        IndexedRouteResponse(
+            routeResponse: mockedResponse(profile: profile),
+            routeIndex: 0
+        )
+    }
+
+    static func mockedResponse(options: NavigationRouteOptions) -> RouteResponse {
+        Fixture.routeResponse(
+            from: "profile-route-original",
+            options: options
+        )
+    }
+
+    static func mockedIndexRouteResponse(
+        options: NavigationRouteOptions
+    ) -> IndexedRouteResponse {
+        IndexedRouteResponse(
+            routeResponse: mockedResponse(options: options),
+            routeIndex: 0
+        )
+    }
+}
+
+fileprivate extension CLLocationCoordinate2D {
+    static var origin: CLLocationCoordinate2D {
+        .init(latitude: -73.98778274913309, longitude: 40.76050975068355)
+    }
+
+    static var destiantion: CLLocationCoordinate2D {
+        .init(latitude: -73.98039053825985, longitude: 40.75988085727627)
+    }
+}
+
+fileprivate extension TimeInterval {
+    static let defaultDelay: Self = 5
+}
+
+fileprivate extension ProfileIdentifier {
+    static let custom: ProfileIdentifier = .init(rawValue: "custom/driving-traffic")
+}
+
+fileprivate extension URLQueryItem {
+    static let customItem: URLQueryItem = .init(name: "foo", value: "bar")
+}
+
+fileprivate final class CustomRouteOptions: NavigationRouteOptions {
```

Comment:
There is already a class CustomRouteOptions in the `TestHelper`. Is it possible to reuse it there somehow?